pkg/benchmark/job/service/option.go

```diff
@@ -0,0 +1,127 @@
+//
+// Copyright (C) 2019-2022 vdaas.org vald team <vald@vdaas.org>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package service manages the main logic of benchmark job.
+package service
+
+import (
+	"github.com/vdaas/vald/internal/client/v1/client/vald"
+	"github.com/vdaas/vald/internal/errgroup"
+	"github.com/vdaas/vald/internal/errors"
+	"github.com/vdaas/vald/internal/test/data/hdf5"
+)
+
+type Option func(j *job) error
+
+var defaultOpts = []Option{
+	WithDimension(748),
+	WithNum(10),
+	WithMinNum(10),
+	WithRadius(-1),
+	WithEpsilon(0.1),
+	WithTimeout("1s"),
+}
+
+func WithDimension(dim int) Option {
+	return func(j *job) error {
+		j.dimension = dim
```

Comment:
Is it better to verity argument ? 🤔 

```go
func WithDimension(dim int) Option {
	return func(j *job) error {
		if dim > 0 {
			j.dim = dim
		}
		return nil
	}
}

```

Reply:
I think it is the same as other functional option implementation.

Reply:
@kevindiu 
yes, but this code is used default option (`WithDimension(748)`). 
so if we use default option value, we need to verify argument. 🤔  If not, the default value would be meaningless.